packages/x-data-grid/src/models/gridSlotsComponentsProps.ts

```diff
@@ -131,6 +149,18 @@ export interface GridSlotProps {
   toolbar: GridToolbarProps & ToolbarPropsOverrides;
 }
 
+/* Merge MUI types into base types to keep slotProps working. */
+type Merge<A, B> = {
```

Comment:
Shouldn't we define this in a more general (utils) space? This seems as if it could be useful elsewhere as well!